### DISCUSSION TITLE
Switch from Google Groups to GitHub Discussions and Import of Google Group Posts

### DISCUSSION THREAD

**Discussion by jeremyevans** [jeremyevans/autoforme#11] [General]
In a couple days, I plan on switching the discussion forum for AutoForme from Google Groups to GitHub Discussions and importing the information from Google Groups to GitHub Discussions.  I'm posting this announcement in case you are watching the repository, because I believe that when the import happens, it will probably generate an email for every discussion and comment added (over 80 emails).  If that will be a problem for you, you may want to unwatch the repository now and then watch it again next week.

**Comment by bpamiri** [comment]
How did this migration go for you. I’m in the same boat. I am the maintainer of an open source project that has been using Google Groups for support and wish to switch to GitHub Discussions. Any thoughts you can share on the process would be appreciated. Also, if you could share any scripts you used to facilitate the migration of messages, that would be most helpful.

**Reply by adam12** [reply]
I used Jeremy's migration script to move the Unpoly Google Group. To prevent one person from owning all the existing content, we created a special @unpolybot account to handle the imports.

A few issues arose. I'm not sure of any workarounds at this point. I brought it up with the Github Discussions PM but I never heard anything back from them.

1. I test imported with my own account and it went fine, but when I did the import with the unpolybot account, it was disabled almost immediately by Github, twice. They continually reactivated it after some time, but Github support seems somewhat of a black hole, and I am never sure if it will take hours or months to resolve a ticket.

2. If there was an `@mention` in the Google Group, ie, `@jeremy`, that person will be notified as if they were mentioned in the post. This is problematic as if you look at `@jeremy`, it's probably not the one that might be the one deserving of mentions. We had a non-related person receive _many_ notifications because of the common first name used. This might be the worst part, and I felt bad for spamming someone because of it.

We asked people to Unwatch the repository before we did the import, to hopefully keep the noise down.

You can somewhat see this playout in my Google Groups thread here: https://groups.google.com/g/unpoly/c/x15bKviN_r4
And the Github Discussion thread here: https://github.com/unpoly/unpoly/discussions/185

**Comment by jeremyevans** [comment]
This is the script I built for the migration: https://github.com/jeremyevans/gg2ghd

If I had to do it over, I would just start using GitHub Discussions, I wouldn't try to migrate content from Google Groups.  That's the approach I followed in all of my larger projects, and it worked fine.

**Comment by bpamiri** [comment]
Thanks for the input guys. We have over 5000 message in the google groups and I think it would be a nightmare to migrate all of them over so I'm also leaning towards just retiring the Google Groups and moving all new discussions over to Discussions. You guys helped confirm some of my doubts.

**Comment by adam12** [comment]
I'd agree that the migration isn't worth the cost. I'd probably just leave a tombstone entry in Google Groups that the discussions have moved.
